### ISSUE TITLE
Incorrect auto-correct Lint/LiteralInInterpolation if contains numbers

### ISSUE THREAD

**Issue by david942j** [bbatsov/rubocop#5862]
Hi rubocop teams,
I found rubocop incorrectly fixes the LiteralInInterpolation.

This bug causes the Ruby program behave different between before/after being fixed by rubocop's auto-correct.

--------

## Expected behavior
```
$ cat a.rb
# puts "Hey, #{0x20}"
$ rubocop a.rb -a
<output omitted>
$ cat a.rb
# puts 'Hey, 32'
```

## Actual behavior

```
$ cat a.rb
# puts "Hey, #{0x20}"
$ rubocop a.rb -a
<output omitted>
$ cat a.rb
# puts 'Hey, 0x20'   <----- incorrect!
```

Invoke rubocop without auto-correct, it shows:
```
Inspecting 1 file
W

Offenses:

a.rb:1:14: W: Lint/LiteralInInterpolation: Literal interpolation detected.
puts "Hey, #{0x20}"
             ^^^^

1 file inspected, 1 offense detected
```

## Steps to reproduce the problem

Prepare `a.rb`:
```ruby
puts "Hey, #{0x20}"
```
then
```
$ rubocop -a a.rb
```

## RuboCop version

Include the output of `rubocop -V`. Here's an example:

```
$ rubocop -V
0.55.0 (using Parser 2.5.1.0, running on ruby 2.4.3 x86_64-linux)
```

**Comment by koic** [comment]
Thanks for the feedback. I confirmed this reproduction. I opened a PR #5863.
